packages/clarity-js/src/layout/extract.ts

```diff
@@ -0,0 +1,64 @@
+import { Extract, Metric, Region, RegionFilter } from "@clarity-types/core";
+import { Constant } from "@clarity-types/data";
+import * as metric from "@src/data/metric";
+import * as region from "@src/layout/region";
+
+const formatRegex = /1/g;
+const digitsRegex = /[^0-9\.]/g;
+const digitsWithCommaRegex = /[^0-9\.,]/g;
+
+export function regions(root: ParentNode, value: Region[]): void {
+    for (let v of value) {
+        let valid = true;
+        let filter = v.length === 4 ? v[2] : null;
+        switch (filter) {
+            case RegionFilter.Url: valid = !!top.location.href.match(new RegExp(v[3])); break;
+            case RegionFilter.Javascript: valid = !!evaluate(v[3]); break;
```

Comment:
>Javascript [](start = 30, length = 10)

how would we handle cases where the js variable is set after an element is parsed?

Reply:
It is triggered on mutations too so we will have more opportunities to look it up.

Reply:
We process this after every mutation (see: dom.ts) - so we will be fine if variable is introduced later in the page.

---
In reply to: [643631150](https://github.com/microsoft/clarity/pull/137#discussion_r643631150) [](ancestors = 643631150)